AntennaTracker/control_auto.cpp

```diff
@@ -58,3 +70,57 @@ void Tracker::calc_body_frame_target(float pitch, float yaw, float& bf_pitch, fl
     bf_pitch   = ahrs.cos_roll()     * pitch    + ahrs.sin_roll()  * ahrs.cos_pitch() * yaw;
     bf_yaw     = -ahrs.sin_roll()    * pitch    + ahrs.cos_pitch() * ahrs.cos_roll()  * yaw;
 }
+bool Tracker::convert_bf_to_ef(float pitch, float yaw, float& ef_pitch, float& ef_yaw)
+{
+    // avoid divide by zero
+    if (is_zero(ahrs.cos_pitch())) {
+        return false;
+    }
+    // convert earth frame angle or rates to body frame
+    ef_pitch =  ahrs.cos_roll()                     * pitch -  ahrs.sin_roll()                     * yaw; //pitch
+    ef_yaw =   (ahrs.sin_roll() / ahrs.cos_pitch()) * pitch + (ahrs.cos_roll() / ahrs.cos_pitch()) * yaw; //yaw
+    return true;
+}
+void Tracker::set_servo_direction(bool& direction_reversed)
```

Comment:
I think set_servo_direction should be merged into calc_angle_error because the set_servo_direction is never called from anywhere else and my impression is that there is some overlap in calculations.  I.e. set_servo_direction calculates "ef_yaw_angle_error" which is really the same as "calc_angle_error"'s "nav_status.angle_error_yaw.